File: 1-ASMTimeConsuming/app/src/main/java/com/study/asm/MainActivity.java
package com.study.asm;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        sleep();
    }

    public void sleep() {
        try {
            Thread.sleep(2_000);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }
}

File: 3-ASM-Booster/app/src/main/java/com/study/test/MainActivity.kt
package com.study.test

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log

class MainActivity : AppCompatActivity() {
    private val TAG = "DOKIT_SLOW_METHOD"

    override fun onCreate(savedInstanceState: Bundle?) {
        val startT = System.currentTimeMillis()
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Thread.sleep(100)
        val endT = System.currentTimeMillis()
        Log.i(TAG, "====Dokit:cost:${endT - startT}ms. in Activity")
    }
}